src/Commands/Domain/SetPrimaryCommand.php

```diff
@@ -0,0 +1,104 @@
+<?php
+
+
+namespace Pantheon\Terminus\Commands\Domain;
+
+use Consolidation\AnnotatedCommand\AnnotationData;
+use Pantheon\Terminus\Commands\TerminusCommand;
+use Pantheon\Terminus\Commands\WorkflowProcessingTrait;
+use Pantheon\Terminus\Models\Environment;
+use Pantheon\Terminus\Models\Site;
+use Pantheon\Terminus\Site\SiteAwareInterface;
+use Pantheon\Terminus\Site\SiteAwareTrait;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+
+/**
+ * Class SetPrimaryCommand
+ * @package Pantheon\Terminus\Commands\Domain
+ */
+class SetPrimaryCommand extends TerminusCommand implements SiteAwareInterface
+{
+    use SiteAwareTrait;
+    use WorkflowProcessingTrait;
+
+    /**
+     * Sets the primary domain for a site and environment, causing all traffic to redirect to the primary domain.
+     *
+     * @authorize
+     *
+     * @command domain:primary:set
+     *
+     * @param string $site_env Site & environment in the format `site-name.env`
+     * @param string $domain A domain that has been associated to your site. Optional when running interactively.
+     *
+     * @usage domain:primary:set <site_env>
+     */
+    public function set($site_env, $domain)
+    {
+        /**
+         * @var $site Site
+         * @var $env Environment
+         */
+        list(, $env) = $this->getSiteEnv($site_env);
+
+        // The primary domain is set via a workflow so as to use workflow logging to track changes.
+        $this->log()->notice('Setting primary domain to {domain}...', ['domain' => $domain]);
+        $workflow = $env->setPrimaryDomain($domain);
```

Comment:
Is there a possibility of verifying that the domain the user entered does actually belong to the current environment? We're already checking for this in the workflow but that's 1) not a good UX (start a workflow in the command line, check for potential errors on the dashboard) and 2) due to the way ygg handles workflow errors, the message itself is not always visible to the customer even on the dash (errors in tasks, for instance, are only visible in the debug tab).

Reply:
I just pulled this on my local and it does show the workflow error message. Nvm :)